lib/Sema/TypeCheckStmt.cpp

```diff
@@ -480,6 +480,7 @@ class StmtChecker : public StmtVisitor<StmtChecker, Stmt*> {
   }
     
   Stmt *visitDeferStmt(DeferStmt *DS) {
+    TC.typeCheckDecl(DS->getTempDecl(), /*isFirstPass*/true);
```

Comment:
I think these are deliberately not doing two passes because they are local, as you said.

Reply:
I'm hoping to get rid of the two passes altogether, but even then if we make two passes over local declarations it shouldn't make much of a difference since it's doing the same amount of work in the end.